### PR TITLE
Dev release documentation and minor changes

### PR DESCRIPTION
**Pull request by birdzix**
This contains documentation changes targeting a developer release.

Also includes some very minor changes to Xi-CAM classes.

### CODE REVIEW

File: docs/source/faq.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# FAQ
+
+## Python
+
+### What is an entry point?
```
Comment (JulReinhardt) [inline docs/source/faq.md]: add link to entry_points.thml

File: docs/source/gui-plugin.md [hunk 2]
```diff
@@ -1,414 +1,344 @@
...
-from xicam.core import msg
-from xicam.core.execution import Workflow
-from xicam.gui.widgets.dynimageview import DynImageView
-from xicam.plugins import GUIPlugin, GUILayout, ProcessingPlugin, Input, Output
+```python
+from databroker import catalog_search_path
+print(catalog_search_path())
+```
 
+You can move the `catalog.yml` file in any of the paths listed.
```
Comment (JulReinhardt) [inline docs/source/gui-plugin.md:295]: Explanation of the difference ("local vs global?") might be desired by potential users.

File: docs/source/operation-plugin.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+the following decorators *must* be used:
+* `@operation` -- allows creation of operations from the function
+* `@output_names` -- defines the name of the output(s)
+
+Additionally, although not required to for an operation,
+the following decorators are highly-recommended for use:
+* `@display_name` -- the name of the operation
+* `@describe_input` -- attach a description to the specified input (can be used multiple times)
+* `@describe_output` -- attach a description to the specified output (can be used multiple times)
+
```
Comment (JulReinhardt) [inline docs/source/operation-plugin.md:87]: plot_hint example would be very useful

File: docs/source/workflow.md [hunk 4]
```diff
@@ -1,108 +1,329 @@
...
+Removing links:
+* `remove_link` -- remove a link from the `Workflow`
+* `clear_operation_links` -- remove all links for a specified operation in the `Workflow`
+* `clear_links` -- remove all links in the `Workflow`
+
+### Enabling and Disabling an Operation
+
+It is possible to enable or disable operations.
+By default, all operations added to a `Workflow` are enabled.
+For more information, see the [API Reference](#api-reference).
```
Comment (JulReinhardt) [inline docs/source/workflow.md:144]: I can't find set_disabled and toggle_disabled in the API ref

**Review by JulReinhardt — COMMENTED** [review]
Discussed a few changes in documentation files with Ian, suggestions have been implemented. Some more bits to add, see comments below.

File: docs/source/workflow.md (hunk 4 above)
Comment (birdzix) [reply docs/source/workflow.md:144]: I think there were issues with the docstrings; I believe that these should be fixed now

File: docs/source/faq.md (hunk 1 above)
Comment (birdzix) [reply docs/source/faq.md]: I temporarily removed this file, there is a separate entry points source file now

File: docs/source/gui-plugin.md (hunk 2 above)
Comment (birdzix) [reply docs/source/gui-plugin.md:295]: That's a good idea; i just added a commit for this

**Pull request merged**